### DISCUSSION TITLE
pixi activation in a docker container

### DISCUSSION THREAD

**Discussion by colindaven** [prefix-dev/pixi#303] [Q&A] [answered]
Dear all, 

I like pixi, it's fast, but I don't know how to activate it easily in a docker container so the installed binaries like Rscript, libraries and R packages can be found.

This is my code - which works.

I create a docker container, then a singularity container from this. I can run a binary such as Rscript if I use the full path in the container, 
eg `/opt/pixi/.pixi/env/bin/Rscript`
but not with just `Rscript`

Am I missing something obvious or has someone got an example?
Thanks

```
# Try out pixi for software env 

RUN apt update && apt install -y curl tar  

RUN mkdir /opt/pixi && cd /opt/pixi && curl -fsSL https://pixi.sh/install.sh | bash

#channel order correct for r packages?
RUN cd /opt/pixi && pixi init --channel conda-forge --channel bioconda && \
  pixi add bioconductor-deseq2==1.34.0 r-dplyr r-ggbeeswarm r-ashr bioconductor-rhdf5 bioconductor-tximport bioconductor-reportingtools r-hwriter r-ini r-ggrepel r-png bioconductor-complexheatmap r-circlize pandas numpy

# how do I activate the pixi env in the container ?
```

**Comment by baszalmstra** [comment, marked as answer]
You cannot globally activate a shell like you can with mamba/micromamba. If you want to run a command in a container you can simply do:

```shell
pixi run <your command goes here>
```

I would recommend adding `pixi install` to the RUN command in your docker to cache the installation of all dependencies.

**Comment by bcm0** [comment]
You could try `SHELL` in the Dockerfile https://docs.docker.com/engine/reference/builder/#shell but I don't know the exact pixi command. I opened a similar discussion for bash scripts https://github.com/prefix-dev/pixi/discussions/690

**Comment by lucascolley** [comment]
closing as resolved, but watch out for a `pixi-docker` extension which may materialise 👀
